### PR TITLE
Support for creating a task that runs a script

### PR DESCRIPTION
**Pull request by richfitz**
More for completeness, but I can imagine that someone will want this at some point.

~Review after #35 is merged, includes those commits~

### CODE REVIEW

File: R/task.R [hunk 1]
```diff
@@ -182,6 +241,7 @@ task_eval <- function(id, envir = .GlobalEnv, verbose = FALSE, root = NULL) {
       data$type,
       explicit = task_eval_explicit(data, envir, verbose, root),
       expression = task_eval_expression(data, envir, verbose, root),
+      script = task_eval_script(data, envir, verbose, root),
```
Comment (weshinsley) [inline R/task.R:244]: Do any of these three calls need `root` ?

**Review by weshinsley — APPROVED** [review]
OK - think this is all good - I had to think a bit carefully about paths on the way through, but I don't think anything needs any changes.

File: R/task.R (hunk 1 above)
Comment (richfitz) [reply R/task.R:244]: seems like not, probably they used to? I can remove that easily enough

File: R/task.R [hunk 2]
```diff
@@ -634,6 +694,19 @@ task_eval_expression <- function(data, envir, verbose, root) {
 }
 
 
+task_eval_script <- function(data, envir, verbose, root) {
```
[the remark this answers is not in the document]
Comment (richfitz) [reply R/task.R:697]: `data$path` is used to set the working directory (that's the old `getwd()` relative to the hipercow root). All task types support/require `path` to exist but none should really reference it when they're called at this point (similar to environment)

File: R/task.R [hunk 3]
```diff
@@ -132,6 +132,65 @@ task_create_expr <- function(expr, environment = "default", submit = NULL,
...
+  root <- hipercow_root(root)
+  if (!file.exists(script)) {
+    cli::cli_abort("Script file '{script}' does not exist")
+  }
+  if (!fs::path_has_parent(script, root$path$root)) {
+    cli::cli_abort(
+      "Script file '{script}' is not contained within hipercow root")
+  }
+  path <- relative_workdir(root$path$root)
+  script <- as.character(fs::path_rel(script, getwd()))
```
[the remark this answers is not in the document]
Comment (richfitz) [reply R/task.R:170]: enforcing, rather than requiring, yes. I wondered about being more clever but decided against it...

**Pull request merged**